llama-index-integrations/embeddings/llama-index-embeddings-gaudi/llama_index/embeddings/gaudi/base.py

```diff
@@ -0,0 +1,277 @@
+# This file is adapted from
+# https://github.com/HabanaAI/Gaudi-tutorials/blob/main/PyTorch/localGPT_inference/gaudi_utils/embeddings.py
+#
+import json
+import os
+from collections import OrderedDict
+
+import numpy as np
+import torch
+from InstructorEmbedding import INSTRUCTOR_Pooling, INSTRUCTOR_Transformer
+from InstructorEmbedding.instructor import batch_to_device, import_from_string
+from sentence_transformers import SentenceTransformer
+from tqdm.autonotebook import trange
+
+from langchain.embeddings import HuggingFaceEmbeddings, HuggingFaceInstructEmbeddings
+
+
+class GaudiSentenceTransformer(SentenceTransformer):
+    """Child class that overrides the tokenize method from SentenceTransformer."""
+
+    def __init__(self, model_name_or_path, embedding_input_size=-1, **kwargs) -> None:
+        super().__init__(model_name_or_path, **kwargs)
+        self.embedding_input_size = embedding_input_size
+
+    def tokenize(self, texts):
+        """Override tokenize method from SentenceTransformer."""
+        return self._first_module().tokenizer(
+            texts,
+            max_length=self.max_seq_length
+            if (
+                self.embedding_input_size == -1
+                or self.embedding_input_size > self.max_seq_length
+            )
+            else self.embedding_input_size,
+            padding="max_length",
+            return_tensors="pt",
+            truncation=True,
+        )
+
+
+class GaudiHuggingFaceEmbeddings(HuggingFaceEmbeddings):
```

Comment:
This is kind of un-intuitive -- can we implement the actual embedding class from llama-index right? No need to involve langhcain (You could subclass BaseEmbedding or the HuggingFaceEmbedding class from llama-index_

Reply:
Okay, I have just re-written the implementation to be based on BaseEmbedding of llama-index. @logan-markewich

Reply:
Thank you!